<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1250</width>
    <height>874</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="image">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>50</y>
      <width>900</width>
      <height>700</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>201</width>
      <height>241</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(204, 204, 204);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>191</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Image Filter</string>
     </property>
    </widget>
    <widget class="QPushButton" name="kernel_convolution_btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>181</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>2D-convolution</string>
     </property>
    </widget>
    <widget class="QPushButton" name="gaussian_btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>181</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Gaussian Blur</string>
     </property>
    </widget>
    <widget class="QPushButton" name="median_btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>181</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Median Blur</string>
     </property>
    </widget>
    <widget class="QPushButton" name="bilateral_btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>160</y>
       <width>181</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Bilateral Blur</string>
     </property>
    </widget>
    <widget class="QPushButton" name="sharpen_btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>181</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Sharpening</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>300</y>
      <width>201</width>
      <height>121</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(204, 204, 204);</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>191</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Edge Detection</string>
     </property>
    </widget>
    <widget class="QPushButton" name="sobel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>181</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Sobel</string>
     </property>
    </widget>
    <widget class="QPushButton" name="canny">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>181</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Canny</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1250</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="select_image_menu"/>
    <addaction name="save_image_menu"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menublur">
    <property name="title">
     <string>Image Filter</string>
    </property>
    <addaction name="kernel_convolution_menu"/>
    <addaction name="gaussian_menu"/>
    <addaction name="median_menu"/>
    <addaction name="bilateral_menu"/>
    <addaction name="sharpen_menu"/>
   </widget>
   <widget class="QMenu" name="menuImage_Edge_Detection">
    <property name="title">
     <string>Edge Detection</string>
    </property>
    <addaction name="sobel_menu"/>
    <addaction name="canny_menu"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menublur"/>
   <addaction name="menuImage_Edge_Detection"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="select_image_menu">
   <property name="text">
    <string>Select Image</string>
   </property>
  </action>
  <action name="actionSave_Result_As">
   <property name="text">
    <string>Save Result As....</string>
   </property>
  </action>
  <action name="actionreplace_original">
   <property name="text">
    <string>replace original</string>
   </property>
  </action>
  <action name="kernel_convolution_menu">
   <property name="text">
    <string>2D-convolution</string>
   </property>
  </action>
  <action name="gaussian_menu">
   <property name="text">
    <string>Gaussian Blur</string>
   </property>
  </action>
  <action name="median_menu">
   <property name="text">
    <string>Median Blur</string>
   </property>
  </action>
  <action name="bilateral_menu">
   <property name="text">
    <string>Bilatera Blur</string>
   </property>
  </action>
  <action name="sharpen_menu">
   <property name="text">
    <string>Sharpening</string>
   </property>
  </action>
  <action name="save_image_menu">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="sobel_menu">
   <property name="text">
    <string>Sobel Edge Detection</string>
   </property>
  </action>
  <action name="canny_menu">
   <property name="text">
    <string>Canny Edge Detection</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
